projects/igniteui-angular/src/lib/grids/grid-navigation.service.ts

```diff
@@ -428,34 +502,43 @@ export class IgxGridNavigationService {
     }
 
     private handleMCHeaderNav(key: string, ctrl: boolean) {
+        const newHeaderNode: ColumnGroupsCache = {
+            visibleIndex: this.activeNode.mchCache.visibleIndex,
+            level: this.activeNode.mchCache.level
+        };
         const activeCol = this.currentActiveColumn;
         const lastGroupIndex = Math.max(... this.grid.visibleColumns.
                 filter(c => c.level <= this.activeNode.level).map(col => col.visibleIndex));
         let nextCol = activeCol;
         if ((key.includes('left') || key === 'home') && this.activeNode.column > 0) {
             const index = ctrl || key === 'home' ? 0 : this.activeNode.column - 1;
             nextCol =  this.getNextColumnMCH(index);
-            this.activeNode.mchCache.visibleIndex = nextCol.visibleIndex;
+            newHeaderNode.visibleIndex = nextCol.visibleIndex;
         }
         if ((key.includes('right') || key === 'end') && activeCol.visibleIndex < lastGroupIndex) {
             const nextVIndex = activeCol.children ? Math.max(...activeCol.allChildren.map(c => c.visibleIndex)) + 1 :
             activeCol.visibleIndex + 1;
             nextCol = ctrl || key === 'end' ? this.getNextColumnMCH(this.lastColumnIndex) : this.getNextColumnMCH(nextVIndex);
-            this.activeNode.mchCache.visibleIndex = nextCol.visibleIndex;
+            newHeaderNode.visibleIndex = nextCol.visibleIndex;
         }
         if (!ctrl && key.includes('up') && this.activeNode.level > 0) {
             nextCol = activeCol.parent;
-            this.activeNode.mchCache.level = nextCol.level;
+            newHeaderNode.level = nextCol.level;
         }
         if (!ctrl && key.includes('down') && activeCol.children) {
-            nextCol = activeCol.children.find(c => c.visibleIndex === this.activeNode.mchCache.visibleIndex) ||
+            nextCol = activeCol.children.find(c => c.visibleIndex === newHeaderNode.visibleIndex) ||
             activeCol.children.toArray().sort((a, b) => b.visibleIndex - a.visibleIndex)
-            .filter(col => col.visibleIndex < this.activeNode.mchCache.visibleIndex)[0];
-            this.activeNode.mchCache.level = nextCol.level;
+            .filter(col => col.visibleIndex < newHeaderNode.visibleIndex)[0];
+            newHeaderNode.level = nextCol.level;
         }
-        this.activeNode.column = nextCol.visibleIndex;
-        this.activeNode.level = nextCol.level;
-        this.performHorizontalScrollToCell(this.activeNode.column);
+
+        this.setActiveNode({
```

Comment:
Also, can we use interfaces for these type of objects, whenever you create an object we should strive to always ensure type safety. As I see, we have a similar `IActiveNode` interface, where gridID and layout are optional.

You've added the logic of Object.assigned to the setActiveNode, which is nice.

Reply:
Actually setActiveNode method does the validation. As you can observe the method validates the arguments itself.